frappe/desk/doctype/form_tour/form_tour.js

```diff
@@ -2,36 +2,64 @@
 // For license information, please see license.txt
 
 frappe.ui.form.on("Form Tour", {
-	setup: function (frm) {
-		if (!frm.doc.is_standard || frappe.boot.developer_mode) {
-			frm.trigger("setup_queries");
-		}
-	},
-
 	refresh(frm) {
 		if (frm.doc.is_standard && !frappe.boot.developer_mode) {
 			frm.trigger("disable_form");
 		}
-
-		frm.add_custom_button(__("Show Tour"), async () => {
-			const issingle = await check_if_single(frm.doc.reference_doctype);
-			let route_changed = null;
-
-			if (issingle) {
-				route_changed = frappe.set_route("Form", frm.doc.reference_doctype);
-			} else if (frm.doc.first_document) {
-				const name = await get_first_document(frm.doc.reference_doctype);
-				route_changed = frappe.set_route("Form", frm.doc.reference_doctype, name);
-			} else {
-				route_changed = frappe.set_route("Form", frm.doc.reference_doctype, "new");
+		frm.fields_dict["report_name"].get_query = function (doc) {
+			if (doc.reference_doctype) {
+				return {
+					filters: {
+						ref_doctype: doc.reference_doctype,
+					},
+				};
 			}
-			route_changed.then(() => {
-				const tour_name = frm.doc.name;
-				cur_frm.tour.init({ tour_name }).then(() => cur_frm.tour.start());
+			return {};
+		};
+		frm.fields_dict["reference_doctype"].get_query = function (doc) {
+			return {
+				filters: {
+					istable: 0,
+				},
+			};
+		};
+		!frm.doc.ui_tour &&
+			frm.add_custom_button(__("Show Tour"), async () => {
+				const issingle = await check_if_single(frm.doc.reference_doctype);
+				let route_changed = null;
+
+				if (issingle) {
+					route_changed = frappe.set_route("Form", frm.doc.reference_doctype);
+				} else if (frm.doc.first_document) {
+					const name = await get_first_document(frm.doc.reference_doctype);
+					route_changed = frappe.set_route("Form", frm.doc.reference_doctype, name);
+				} else {
+					route_changed = frappe.set_route("Form", frm.doc.reference_doctype, "new");
+				}
+				route_changed.then(() => {
+					const tour_name = frm.doc.name;
+					cur_frm.tour.init({ tour_name }).then(() => cur_frm.tour.start());
+				});
 			});
-		});
 	},
-
+	async report_name(frm) {
+		if (!frm.doc.report_name) return;
+		let { message } = await frappe.db.get_value("Report", frm.doc.report_name, "ref_doctype");
+		frm.set_value("reference_doctype", message?.ref_doctype || "");
+	},
+	async before_save(frm) {
+		if (
+			frm.doc.select_view == "List" &&
+			frm.doc.list_name == "Dashboard" &&
+			frm.doc.dashboard_name &&
+			frm.doc.reference_doctype
+		) {
+			frappe.throw(
+				"Referance Doctype and Dashboard Name both can't be used at the same time."
+			);
+		}
```

Comment:
Avoid throwing error, clear & hide the other field if one field is set

Reply:
We don't know which one user intends to use  and there isn't always a relation between two. that's why i am trowing error instead to clearing.